### DISCUSSION TITLE
When is it necessary to use alternative YAML syntax?

### DISCUSSION THREAD

**Discussion by ghost** [squidfunk/mkdocs-material#7013] [Q&A]
Hi!

I am using configuration inheritance in my mini-project. According to the documentation (https://www.mkdocs.org/user-guide/configuration/#configuration-inheritance ), it says that the markdown_extensions and plugins parameters should use an alternative syntax that does not use list items (lines starting with -). However, this syntax is still being used in some places, such as in themes and in the watch command. As a result, I was confused about how to use the new syntax. Should I use it only in the specified parameters, or should I use it for all of them?

features:
    - navigation.indexes
    - navigation.instant
    - toc.follow
    - search.highlight
    - content.action.edit
    - content.action.view

**Thanks!**

**Comment by alexvoss** [comment]
I can only hazard a guess as to why MkDocs requires the alternative syntax: to be able to merge the two configurations more easily, being able to look up by name instead of iterating through a list.

Purely from a YAML point of view, you can use *either* a list *or* key/value pairs, not both. Think of `features` as being either a list or a dictionary. It cannot be both at the same time.

**Comment by ghost** [comment]
Thank you very much. Then, I will only use what I need for plugins and extensions.
